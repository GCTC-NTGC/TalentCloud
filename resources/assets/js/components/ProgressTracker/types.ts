/* eslint-disable import/prefer-default-export */

export type ProgressTrackerState = "active" | "complete" | "error" | "null";

export interface ProgressTrackerItem {
  state: ProgressTrackerState;
  label: string;
  title: string;
}
