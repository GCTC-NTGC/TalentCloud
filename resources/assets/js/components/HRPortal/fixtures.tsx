import { text, boolean, number } from "@storybook/addon-knobs";
import { JobAction } from "../../models/types";
import { JobStatus } from "../../models/lookupConstants";

export const jobActions: JobAction[] = [
  {
    activity: {
      count: 3,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 5,
    classification: "CS01",
    draft: {
      url: "https://google.ca",
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: true,
    preview: {
      url: "https://google.ca",
      text: "Preview Poster",
      title: "Preview what this Job Poster will look like.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Draft,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Front-end Developer",
    userTime: 2,
  },
  {
    activity: {
      count: 0,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 12,
    classification: "BK01",
    draft: {
      url: null,
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: false,
    preview: {
      url: "https://google.ca",
      text: "View Poster",
      title: "View the published Job Poster.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Published,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Cupcake Baker",
    userTime: 2,
  },
  {
    activity: {
      count: 0,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 5,
    classification: "BS01",
    draft: {
      url: null,
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: false,
    preview: {
      url: "https://google.ca",
      text: "View Poster",
      title: "View the published Job Poster.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Complete,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Underwater Basket Weaver",
    userTime: 2,
  },
];

export const unclaimedJobs = [
  {
    title: "CS01 - Front-end Developer",
    createdAt: "2019-MAY-02",
    status: JobStatus.Draft,
    hiringManagers: ["Rebecca Appleby"],
    hrAdvisors: [],
  },
  {
    title: "AS02 - Executive Assisstant",
    createdAt: "2019-MAR-12",
    status: JobStatus.Draft,
    hiringManagers: ["Rebecca Appleby"],
    hrAdvisors: ["Rebecca Appleby", "Jack Little"],
  },
  {
    title: "ET03 - Business Analyst",
    createdAt: "2019-DEC-02",
    status: JobStatus.Review,
    hiringManagers: ["Robin Browne"],
    hrAdvisors: [],
  },
  {
    title: "FG05 - Long-term Care Specialist",
    createdAt: "2019-MAY-14",
    status: JobStatus.Draft,
    hiringManagers: ["Braeden McDoogal"],
    hrAdvisors: ["Caitlyn Summers", "Jack Little"],
  },
  {
    title: "CS03 - Digital Product Designer",
    createdAt: "2019-JUL-24",
    status: JobStatus.Review,
    hiringManagers: ["Amelie Lachance"],
    hrAdvisors: [],
  },
];
