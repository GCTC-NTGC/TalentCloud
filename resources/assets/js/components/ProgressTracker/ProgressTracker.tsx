import React from "react";
import { ProgressTrackerItem } from "./types";
import ProgressTrackerItemComponent from "./ProgressTrackerItem";

export interface ProgressTrackerProps {
  items?: ProgressTrackerItem[];
  children?: React.ReactNode;
  backgroundColor?: string;
  backgroundOpacity?: string;
  fontColor?: string;
  classNames?: string;
  itemsWrapperClassNames?: string;
  dataIsLoading?: boolean;
}

const ProgressTracker: React.FunctionComponent<ProgressTrackerProps> = ({
  items,
  children,
  backgroundColor,
  backgroundOpacity,
  fontColor,
  classNames,
  itemsWrapperClassNames,
  dataIsLoading,
}): React.ReactElement => {
  return (
    <div
      data-c-alignment="base(centre)"
      data-c-background={`${backgroundColor}(${backgroundOpacity || 100})`}
      className={classNames}
    >
      <div
        data-c-container="layout"
        data-c-padding="top(quarter) bottom(quarter)"
      >
        <div className={itemsWrapperClassNames}>
          {/* If items list exists, then return list of progress tracker item components. Also, progress tracker items can be passed down to the children props, and will be printed out below. */}
          {items &&
            items.map(
              (item): React.ReactElement => {
                const { state, label, title } = item;
                return (
                  <ProgressTrackerItemComponent
                    key={title}
                    state={state}
                    label={label}
                    title={title}
                    fontColor={fontColor}
                    dataIsLoading={dataIsLoading}
                  />
                );
              },
            )}
          {children}
        </div>
      </div>
    </div>
  );
};

export default ProgressTracker;
