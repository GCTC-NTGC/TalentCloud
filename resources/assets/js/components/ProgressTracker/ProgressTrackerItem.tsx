import React from "react";

interface ProgressTrackerItemProps {
  state: "active" | "complete" | "error" | "null";
  label: string;
  title: string;
  fontColor?: string;
}
const ProgressTrackerItem: React.FunctionComponent<
  ProgressTrackerItemProps
> = ({ label, title, state, fontColor }): React.ReactElement => {
  return (
    <div
      className="tracker-item"
      data-tc-tracker-state={state}
      data-c-alignment="base(left)"
      data-c-grid="middle"
      data-c-margin="top(half) right(normal) bottom(half) left(normal)"
      data-c-color={fontColor}
    >
      <div className="tracker-icon">
        {state === "active" && <i className="fas fa-arrow-down" />}
        {state === "complete" && <i className="fas fa-check" />}
        {state === "error" && <i className="fas fa-exclamation-triangle" />}
      </div>
      <div className="tracker-title">
        <span data-c-font-size="small">{label}</span>
        <span data-c-font-weight="bold">{title}</span>
      </div>
    </div>
  );
};

export default ProgressTrackerItem;
