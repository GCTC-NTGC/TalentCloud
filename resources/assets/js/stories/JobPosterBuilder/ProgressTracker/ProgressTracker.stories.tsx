import React from "react";
import { storiesOf } from "@storybook/react";
import { text } from "@storybook/addon-knobs";
import { withIntl } from "storybook-addon-intl";
import ProgressTracker from "../../../components/ProgressTracker/ProgressTracker";
import { ProgressTrackerItem } from "../../../components/ProgressTracker/types";

const items: ProgressTrackerItem[] = [
  { link: "/", state: "active", label: "Step 01", title: "Job Info" },
  { link: "/", state: "complete", label: "Step 02", title: "Work Env." },
  { link: "/", state: "error", label: "Step 03", title: "Impact" },
  { link: "/", state: "null", label: "Step 04", title: "Tasks" },
  { link: "/", state: "null", label: "Step 05", title: "Skills" },
  { link: "/", state: "null", label: "Step 05", title: "Review" },
];

const stories = storiesOf(
  "Job Poster Builder|Progress Tracker",
  module,
).addDecorator(withIntl);

stories.add(
  "Tracker",
  (): React.ReactElement => (
    <ProgressTracker
      items={items}
      backgroundColor={text("Background Color", "black")}
      fontColor={text("Font Color", "white")}
      classNames="manager-jpb-tracker"
      itemsWrapperClassNames="tracker manager-jpb-tracker-wrapper"
    />
  ),
);
