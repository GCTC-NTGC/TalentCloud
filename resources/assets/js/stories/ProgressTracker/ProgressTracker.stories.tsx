import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withKnobs, text } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import ProgressTracker from "../../components/ProgressTracker/ProgressTracker";
import { ProgressTrackerItem } from "../../components/ProgressTracker/types";

const items: ProgressTrackerItem[] = [
  { state: "active", label: "Step 01", title: "Job Info" },
  { state: "complete", label: "Step 02", title: "Work Env." },
  { state: "error", label: "Step 03", title: "Impact" },
  { state: "null", label: "Step 04", title: "Tasks" },
  { state: "null", label: "Step 05", title: "Skills" },
  { state: "null", label: "Step 05", title: "Review" },
];

const stories = storiesOf("ProgressTracker", module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs);

stories.add(
  "Progress Tracker",
  (): React.ReactElement => (
    <ProgressTracker
      items={items}
      backgroundColor="black"
      fontColor="white"
      classNames="manager-jpb-tracker"
      itemsWrapperClassNames="tracker manager-jpb-tracker-wrapper"
    />
  ),
  { info: { inline: true } },
);
