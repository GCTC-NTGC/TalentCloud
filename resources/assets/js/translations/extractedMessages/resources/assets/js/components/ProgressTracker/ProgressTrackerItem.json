[{"id":"progressTracker.unreachableStep","description":"Tooltip informing user to complete previous steps, before step it can become clickable.","defaultMessage":"Must complete previous steps."}]
