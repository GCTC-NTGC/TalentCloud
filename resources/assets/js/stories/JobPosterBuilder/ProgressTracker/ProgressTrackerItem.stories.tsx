import React from "react";
import { storiesOf } from "@storybook/react";
import { text, select } from "@storybook/addon-knobs";
import ProgressTrackerItem from "../../../components/ProgressTracker/ProgressTrackerItem";
import ProgressTracker from "../../../components/ProgressTracker/ProgressTracker";

const ProgressTrackerWrapper = (
  children: React.ReactElement,
): React.ReactElement => {
  return (
    <ProgressTracker
      backgroundColor={text("Background Color", "white")}
      itemsWrapperClassNames="tracker"
    >
      {children}
    </ProgressTracker>
  );
};

const stories = storiesOf("Job Poster Builder|Progress Tracker/Item", module);

const trackerStateOptions: {
  [key: string]: "active" | "complete" | "error" | "null";
} = {
  active: "active",
  complte: "complete",
  error: "error",
  null: "null",
};

stories.add(
  "Active",
  (): React.ReactElement =>
    ProgressTrackerWrapper(
      <ProgressTrackerItem
        state={select("State", trackerStateOptions, "active")}
        label={text("Label", "Step 01")}
        title={text("Title", "Job Info")}
        fontColor={text("Font Color", "black")}
      />,
    ),
);

stories.add(
  "Complete",
  (): React.ReactElement =>
    ProgressTrackerWrapper(
      <ProgressTrackerItem
        state={select("State", trackerStateOptions, "active")}
        label={text("Label", "Step 02")}
        title={text("Title", "Work Env.")}
        fontColor={text("Font Color", "black")}
      />,
    ),
);

stories.add(
  "Error",
  (): React.ReactElement =>
    ProgressTrackerWrapper(
      <ProgressTrackerItem
        state={select("State", trackerStateOptions, "active")}
        label={text("Label", "Step 03")}
        title={text("Title", "impact")}
        fontColor={text("Font Color", "black")}
      />,
    ),
);

stories.add(
  "Null",
  (): React.ReactElement =>
    ProgressTrackerWrapper(
      <ProgressTrackerItem
        state={select("State", trackerStateOptions, "active")}
        label={text("Label", "Step 04")}
        title={text("Title", "Tasks")}
        fontColor={text("Font Color", "black")}
      />,
    ),
);
