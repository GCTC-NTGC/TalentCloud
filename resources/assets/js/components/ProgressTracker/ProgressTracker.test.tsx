import React from "react";
import { shallow } from "enzyme";
import ProgressTracker from "./ProgressTracker";
import ProgressTrackerItem from "./ProgressTrackerItem";
import { items } from "./fixtures/progressItems";

describe("Progress Tracker", (): void => {
  it("should render ProgressTracker with items", (): void => {
    const wrapper = shallow(<ProgressTracker items={items} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render ProgressTracker with children", (): void => {
    const wrapper = shallow(
      <ProgressTracker>
        <ProgressTrackerItem {...items[0]} />
        <ProgressTrackerItem {...items[1]} />
        <ProgressTrackerItem {...items[2]} />
      </ProgressTracker>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render ProgressTracker with background color and opacity", (): void => {
    const wrapper = shallow(
      <ProgressTracker backgroundColor="black" backgroundOpacity="100">
        <ProgressTrackerItem {...items[0]} />
      </ProgressTracker>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
