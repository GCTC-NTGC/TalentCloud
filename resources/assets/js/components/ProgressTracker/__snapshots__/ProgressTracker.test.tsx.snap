// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Tracker should render ProgressTracker with background color and opacity 1`] = `
<div
  data-c-alignment="base(centre)"
  data-c-background="black(100)"
  data-c-padding="top(quarter) bottom(quarter)"
>
  <ol
    aria-valuemin={1}
    aria-valuenow={-1}
    aria-valuetext="function () {
        var active = items &&
            items.find(function (_a) {
                var state = _a.state;
                return state === \\"active\\";
            });
        return active ? active.label : \\"\\";
    } : function () {
        var active = items &&
            items.find(function (_a) {
                var state = _a.state;
                return state === \\"active\\";
            });
        return active ? active.title : \\"\\";
    }"
    data-c-container="layout"
    role="progressbar"
    tabIndex={0}
  >
    <ProgressTrackerItem
      label="Step 01"
      state="active"
      title="Job Info"
    />
  </ol>
</div>
`;

exports[`Progress Tracker should render ProgressTracker with children 1`] = `
<div
  data-c-alignment="base(centre)"
  data-c-background="undefined(100)"
  data-c-padding="top(quarter) bottom(quarter)"
>
  <ol
    aria-valuemin={1}
    aria-valuenow={-1}
    aria-valuetext="function () {
        var active = items &&
            items.find(function (_a) {
                var state = _a.state;
                return state === \\"active\\";
            });
        return active ? active.label : \\"\\";
    } : function () {
        var active = items &&
            items.find(function (_a) {
                var state = _a.state;
                return state === \\"active\\";
            });
        return active ? active.title : \\"\\";
    }"
    data-c-container="layout"
    role="progressbar"
    tabIndex={0}
  >
    <ProgressTrackerItem
      label="Step 01"
      state="active"
      title="Job Info"
    />
    <ProgressTrackerItem
      label="Step 02"
      state="complete"
      title="Work Env."
    />
    <ProgressTrackerItem
      label="Step 03"
      state="error"
      title="Impact"
    />
  </ol>
</div>
`;

exports[`Progress Tracker should render ProgressTracker with items 1`] = `
<div
  data-c-alignment="base(centre)"
  data-c-background="undefined(100)"
  data-c-padding="top(quarter) bottom(quarter)"
>
  <ol
    aria-valuemax={3}
    aria-valuemin={1}
    aria-valuenow={1}
    aria-valuetext="function () {
        var active = items &&
            items.find(function (_a) {
                var state = _a.state;
                return state === \\"active\\";
            });
        return active ? active.label : \\"\\";
    } : function () {
        var active = items &&
            items.find(function (_a) {
                var state = _a.state;
                return state === \\"active\\";
            });
        return active ? active.title : \\"\\";
    }"
    data-c-container="layout"
    role="progressbar"
    tabIndex={0}
  >
    <ProgressTrackerItem
      key="Job Info"
      label="Step 01"
      state="active"
      title="Job Info"
    />
    <ProgressTrackerItem
      key="Work Env."
      label="Step 02"
      state="complete"
      title="Work Env."
    />
    <ProgressTrackerItem
      key="Impact"
      label="Step 03"
      state="error"
      title="Impact"
    />
  </ol>
</div>
`;
